### PR TITLE
Fix OIDC sync #428

### PR DESCRIPTION
**Pull request by dciangot**
Image tested in integration server

- [ ] to be pushed into harbor
- [ ] to be included in flux

### CODE REVIEW

**Comment by ericvaandering** [comment]
I don't think any flux changes are needed, right? At best I see that there is a secret you no longer need.

**Comment by dciangot** [comment]
Right, since it is pointing to the latest tag I suppose.

**Comment by dciangot** [comment]
@ericvaandering how do you propose to proceed with this then?

**Comment by ericvaandering** [comment]
Well, I asked questions and for changes in the comments. So let's start by addressing those.

File: docker/CMSRucioClient/Dockerfile [hunk 1]
```diff
@@ -58,18 +60,17 @@ RUN git clone https://github.com/rucio/rucio.git
 COPY scripts /scripts
 COPY loadtest /loadtest
 
+ENV test=test
```
[the remark this answers is not in the document]
Comment (dciangot) [reply docker/CMSRucioClient/Dockerfile]: nothing, removed

File: docker/CMSRucioClient/Dockerfile [hunk 2]
```diff
@@ -58,18 +60,17 @@ RUN git clone https://github.com/rucio/rucio.git
...
 COPY rucio-testbed.cfg /opt/rucio-testbed/etc/rucio.cfg
 COPY rucio-dev.cfg /opt/rucio-dev/etc/rucio.cfg
 COPY rucio-int.cfg /opt/rucio-int/etc/rucio.cfg
 COPY rucio-prod.cfg /opt/rucio-prod/etc/rucio.cfg
 
 # Pull some useful stuff out of git
 WORKDIR /root
-RUN git clone https://github.com/dmwm/CMSRucio.git
+#RUN git clone https://github.com/dmwm/CMSRucio.git
+RUN git clone -b fix_oidc https://github.com/dciangot/CMSRucio.git
```
[the remark this answers is not in the document]
Comment (dciangot) [reply docker/CMSRucioClient/Dockerfile]: right, reverted

File: docker/CMSRucioClient/scripts/k8s_sync_users_links.sh [hunk 3]
```diff
@@ -36,19 +36,5 @@ fi
 echo "Creating links"
 ./cmslinks.py --overwrite --disable
 
-# Preparing oidc-agent to get/refresh token to retrieve OIDC user information
```
[the remark this answers is not in the document]
Comment (dciangot) [reply docker/CMSRucioClient/scripts/k8s_sync_users_links.sh:39]: I works following a credentials authentication flows. Enabled at IAM level, so we do not need oidc-agent anymore for this kind of task

File: helm/rucio-cron-jobs/templates/cronjobs.yaml [hunk 4]
```diff
@@ -113,11 +113,5 @@ spec:
                       name: "rucio-secrets"
                       key: oidc_iam_client_secret
                       optional: true
-                - name: REFRESH_TOKEN
```
[the remark this answers is not in the document]
Comment (dciangot) [reply helm/rucio-cron-jobs/templates/cronjobs.yaml:116]: yes, like above

**Comment by dciangot** [comment]
The secrets needed still the same as https://github.com/dmwm/rucio-flux/tree/main/scripts/create_iam_clients but with different privileges at the level of oidc client in the IAM server. 
The integration one should be already configured properly. Once cross checked we can look at updating the one used for production.

**Comment by ericvaandering** [comment]
Is this ready? Have the secrets been installed in the servers?

**Comment by dciangot** [comment]
@ericvaandering right, correct client id just commited, the rest is in the secret already in the cluster

**Pull request merged**